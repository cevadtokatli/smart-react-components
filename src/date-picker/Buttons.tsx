import React from "react"
import {Value, Language} from "../types/date-picker"
import {SetValue} from "../props"
import Div from "../element/Div"
import Button from "../button/Button"

interface Props {
    value: Value
    setValue: SetValue<Value>
    initialValue: Value
    setInitialValue: React.Dispatch<React.SetStateAction<Value>>
    onSave: () => void
    onCancel: () => void
    type: string
    lang: Language
}

const Buttons: React.FC<Props> = ({value,setValue,initialValue,setInitialValue,onSave,onCancel,type,lang}) => {
    /**
     * Sets initial value as value.
     */
    const cancel = () => {
        setValue(initialValue)
        if(onCancel)
            onCancel()
    }

    /**
     * Sets value as initial value.
     */
    const save = () => {
        setInitialValue(value)
        if(onSave)
            onSave()
    }

    return (
        <Div className="src-date-picker-buttons" display="flex" justifyContent="flex-end" alignItems="center">
            <Button type={type} link={true} waveEffect="dark" elementProps={{onClick:cancel}}>{lang.cancel}</Button>
            <Button type={type} link={true} waveEffect="dark" elementProps={{onClick:save}}>{lang.save}</Button>
        </Div>
    )
}

export default Buttons