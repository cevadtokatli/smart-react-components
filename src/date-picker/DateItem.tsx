import styled from "styled-components"

interface Props {
    inactive: boolean
    active?: boolean
}

export default styled.li.attrs<Props>(({className,active,inactive}) => ({
    className: `${className} src-date-picker-item ${active ? "src-date-picker-item-active" : ""} ${inactive ? "src-date-picker-item-inactive" : ""}`
}))<Props>`
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 100%;
    line-height: 100%;
    font-weight: 600;
`