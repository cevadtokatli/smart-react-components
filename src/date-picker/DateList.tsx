import styled from "styled-components"

export default styled.ul`
    display: flex;
    justify-content: space-between;
    padding: 0;
    margin: 0;
`
