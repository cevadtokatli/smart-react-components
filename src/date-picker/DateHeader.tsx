import React from "react"
import styled from "styled-components"
import size from "../constants/date-picker/size"
import {SelectedDateObject, Language} from "../types/date-picker"
import * as DP from "../constants/date-picker/date-picker-type"
import {MAX_YEAR, MIN_YEAR} from "../constants/date-picker/limit"
import Div from "../element/Div"
import IconContainer from "./IconContainer"
import ArrowUp from "./icons/ArrowUp"
import ArrowDown from "./icons/ArrowDown"

const HeaderElement = styled.header.attrs({
    className: "src-date-picker-date-header"
})`
    display: flex;
    justify-content: space-between;

    > span {
        cursor: default;
        font-weight: 600;
    }

    svg {
        width: ${size.dateHeaderIconSize};
    }
`

interface Props {
    date: SelectedDateObject
    setDate: React.Dispatch<React.SetStateAction<SelectedDateObject>>
    selectedDatePicker: number
    setSelectedDatePicker: React.Dispatch<React.SetStateAction<number>>
    orientation: boolean
    lang: Language
}

const DateHeader: React.FC<Props> = ({date,setDate,selectedDatePicker,setSelectedDatePicker,orientation,lang}) => {
    const getValue = (): string => {
        if(selectedDatePicker == DP.WEEK)
            return `${lang.months[date.month]} ${date.year}`
        else if(selectedDatePicker == DP.YEAR)
            return `${date.year}`
        else
            return `${date.decade[0]} - ${date.decade[1]}`
    }

    const [value, setValue] = React.useState<string>(() => getValue())

    React.useEffect(() => setValue(getValue()), [date, selectedDatePicker])

    /**
     * Sets "selected date picker"
     */
    const setSelectedDatePicker$ = () => {
        if(selectedDatePicker == DP.WEEK)
            setSelectedDatePicker(DP.YEAR)
        else if(selectedDatePicker == DP.YEAR)
            setSelectedDatePicker(DP.DECADE)
    }

    /**
     * Sets date
     * 
     * @param up 
     */
    const setDate$ = (up:boolean) => {
        const inc = orientation ? up : !up
        const val = {...date}

        if(inc) {
            if(selectedDatePicker == DP.WEEK) {
                if(val.month < 11)
                    val.month++
                else if(val.year < MAX_YEAR) {
                    val.month = 0
                    val.year++
                }
            } else if(selectedDatePicker == DP.YEAR) {
                if(val.year < MAX_YEAR)
                    val.year++
            } else if(selectedDatePicker == DP.DECADE) {
                if(val.decade[1] < MAX_YEAR)
                    val.decade = [val.decade[1], val.decade[1]+10]
            }
        } else {
            if(selectedDatePicker == DP.WEEK) {
                if(val.month > 0)
                    val.month--
                else if(val.year > MIN_YEAR) {
                    val.month = 11
                    val.year--
                }
            } else if(selectedDatePicker == DP.YEAR) {
                if(val.year > MIN_YEAR)
                    val.year--
            } else if(selectedDatePicker == DP.DECADE) {
                if(val.decade[0] > MIN_YEAR)
                    val.decade = [val.decade[0]-10, val.decade[0]]
            }
        }

        setDate(val)
    }

    return (
        <HeaderElement>
            <span className="src-date-picker-span-element" onClick={setSelectedDatePicker$}>{value}</span>
            <Div display="flex" alignItems="center">
                <IconContainer>
                    <ArrowUp onClick={() => setDate$(true)} />
                </IconContainer>
                <IconContainer>
                    <ArrowDown onClick={() => setDate$(false)} />
                </IconContainer>
            </Div>
        </HeaderElement>
    )
}

export default DateHeader