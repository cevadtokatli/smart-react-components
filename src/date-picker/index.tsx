import * as DT from "../constants/date-picker/date-type"
import * as HT from "../constants/date-picker/hour-type"
import * as PT from "../constants/date-picker/picker-type"
import * as W from "../constants/date-picker/week-start-day"

export const DATE_TYPE = DT
export const HOUR_TYPE = HT
export const PICKER_TYPE = PT
export const WEEK_START_DAY = W
export {DateHourObject, Language} from "../types/date-picker"
export * from "./DatePicker"
export {default as default} from "./DatePicker"