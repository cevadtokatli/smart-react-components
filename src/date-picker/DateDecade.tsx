import React from "react"
import {SelectedDateObject} from "../types/date-picker"
import DateList from "./DateList"
import DateItem from "./DateItem"
import {MIN_YEAR, MAX_YEAR} from "../constants/date-picker/limit"
import * as DPT from "../constants/date-picker/date-picker-type"

interface Props {
    date: SelectedDateObject
    setDate: React.Dispatch<React.SetStateAction<SelectedDateObject>>
    setSelectedDatePicker: React.Dispatch<React.SetStateAction<number>>
}

const DateDecade: React.FC<Props> = ({date,setDate,setSelectedDatePicker}) => {
    const getList = () => {
        const result = Array(4).fill(null).map((item,index) => React.createElement(DateList, {key:index,children:[]}))
        let index = 0

        for(let i=0; i<16; i++) {
            const year = date.decade[0] + i
            const isInactive = year >= date.decade[1]
            const isActive = year == date.year
            result[index].props.children.push(<DateItem className="src-date-picker-date-item" key={i} inactive={isInactive} active={isActive} onClick={() => setValue(year)}>{year}</DateItem>)
            if(i % 4 == 3)
                index++
        }

        return result
    }

    const [list, setList] = React.useState<JSX.Element[]>(() => getList())
    const listInit = React.useRef<boolean>(false)

    React.useEffect(() => {
        if(!listInit.current)
            listInit.current = true
        else
            setList(getList())
    }, [date])

    /**
     * Sets date.
     * 
     * @param year 
     */
    const setValue = (year:number) => {
        if(year >= MIN_YEAR && year <= MAX_YEAR) {
            setSelectedDatePicker(DPT.YEAR)
            setDate({...date,year})
        }
    }

    return <div>{list}</div>
}

export default DateDecade