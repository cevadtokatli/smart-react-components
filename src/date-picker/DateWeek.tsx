import React from "react"
import {DateHourObject, SelectedDateObject, DateObject, HourObject, Language} from "../types/date-picker"
import DateList from "./DateList"
import DatePickerHelper from "../helper/DatePickerHelper"
import DateItem from "./DateItem"
import {MIN_YEAR, MAX_YEAR} from "../constants/date-picker/limit"

interface Props {
    valueObject: DateHourObject
    date: SelectedDateObject
    setValue: (date:DateObject, hour:HourObject) => void
    weekStartDay: number
    lang: Language
}

const DateWeek: React.FC<Props> = ({valueObject,date,setValue,weekStartDay,lang}) => {
    const getHeader = () => {
        const weekDays = DatePickerHelper.getWeekDays(weekStartDay)
        const result = []
        weekDays.forEach(item => result.push(<DateItem className="src-date-picker-week-item" key={item} inactive={true}>{lang.days[item].substring(0, 2)}</DateItem>))
        return result
    }

    const getList = () => {
        const result = Array(6).fill(null).map((item,index) => React.createElement(DateList, {key:index,children:[]}))

        const prevMonth = date.month == 0 ? 11 : date.month - 1
        const prevYear = prevMonth == 11 ? date.year - 1 : date.year
        const nextMonth = date.month < 11 ? date.month + 1 : 1
        const nextYear = nextMonth == 0 ? date.year + 1 : date.year

        const weekDays = DatePickerHelper.getWeekDays(weekStartDay)
        const firstMonthDay = new Date(date.year, date.month, 1).getDay()
        const weekStartIndex = weekDays.indexOf(firstMonthDay)
        const weekDayCount = DatePickerHelper.getDayCount(date.month, date.year)
        const prevDayLimit = firstMonthDay - weekStartIndex
        const prevMountCount = DatePickerHelper.getDayCount(prevMonth, prevYear)

        let prevDay = prevMountCount - prevDayLimit
        let weekDay = 1
        let nextWeekDay = 1
        let index = 0

        for(let i=0; i<42; i++) {
            let dateItem
            let isInactive

            if(prevDay <= prevMountCount) {
                dateItem = {
                    day: prevDay,
                    month: prevMonth,
                    year: prevYear
                }
                prevDay++
                isInactive = true
            } else if(weekDay <= weekDayCount) {
                dateItem = {
                    day: weekDay,
                    month: date.month,
                    year: date.year
                }
                weekDay++
            } else {
                dateItem = {
                    day: nextWeekDay,
                    month: nextMonth,
                    year: nextYear
                }
                nextWeekDay++
                isInactive = true
            }

            let isActive
            if(
                valueObject &&
                dateItem.day == valueObject.date.day &&
                dateItem.month == valueObject.date.month &&
                dateItem.year == valueObject.date.year
            ) 
                isActive = true
            else
                isActive = false
        
            result[index].props.children.push(<DateItem className="src-date-picker-week-item" key={i} active={isActive} inactive={isInactive} onClick={() => setValue$(dateItem.day, dateItem.month, dateItem.year)}>{dateItem.day}</DateItem>)

            if(i % 7 == 6)
                index++
        }

        return result
    }

    const [header, setHeader] = React.useState<JSX.Element[]>(() => getHeader())
    const headerInit = React.useRef<boolean>(false)
    const [list, setList] = React.useState<JSX.Element[]>(() => getList())
    const listInit = React.useRef<boolean>(false)

    React.useEffect(() => {
        if(!headerInit.current)
            headerInit.current = true
        else
            setHeader(getHeader())
    }, [lang, weekStartDay])

    React.useEffect(() => {
        if(!listInit.current)
            listInit.current = true
        else
            setList(getList())
    }, [valueObject, date])

    /**
     * Sets value.
     * 
     * @param day
     * @param month 
     * @param year 
     */
    const setValue$ = (day:number, month:number, year:number) => {
        if(year >= MIN_YEAR && year <= MAX_YEAR)
            setValue({day,month,year}, null)
    }

    return (
        <div>
            <DateList>{header}</DateList>
            {list}
        </div>
    )
}

export default DateWeek