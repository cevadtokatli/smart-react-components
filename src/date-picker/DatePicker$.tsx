import React from "react"
import styled from "styled-components"
import size from "../constants/date-picker/size"
import {DateHourObject, SelectedDateObject, DateObject, HourObject, Language} from "../types/date-picker"
import DateHeader from "./DateHeader"
import CSSTransition from "../transition/CSSTransition"
import * as DPT from "../constants/date-picker/date-picker-type"
import DateWeek from "./DateWeek"
import DateYear from "./DateYear"
import DateDecade from "./DateDecade"
import TransitionSwitch from "../transition/TransitionSwitch"

const PickerElement = styled.div`
    height: 100%;
    padding: 0 ${size.paddingX};
`

interface Props {
    valueObject: DateHourObject
    date: SelectedDateObject
    setDate: React.Dispatch<React.SetStateAction<SelectedDateObject>>
    selectedDatePicker: number
    setSelectedDatePicker: React.Dispatch<React.SetStateAction<number>>
    setValue: (date:DateObject, hour:HourObject) => void
    orientation: boolean
    weekStartDay: number
    lang: Language
}

const DatePickerItem: React.FC<Props> = ({valueObject,date,setDate,selectedDatePicker,setSelectedDatePicker,setValue,orientation,weekStartDay,lang}) => {
    const previousSelectedDatePicker = React.useRef<number>(null)

    React.useEffect(() => {
        previousSelectedDatePicker.current = selectedDatePicker
    }, [selectedDatePicker])

    return (
        <PickerElement>
            <DateHeader date={date} setDate={setDate} selectedDatePicker={selectedDatePicker} setSelectedDatePicker={setSelectedDatePicker} orientation={orientation} lang={lang} />
            <div>
                <TransitionSwitch active={selectedDatePicker}>
                    <CSSTransition value={DPT.WEEK} className="src-dp-fade-in" duration={150}>
                        <DateWeek valueObject={valueObject} date={date} setValue={setValue} weekStartDay={weekStartDay} lang={lang} />
                    </CSSTransition>
                    <CSSTransition value={DPT.YEAR} className={`${(previousSelectedDatePicker.current == DPT.WEEK || selectedDatePicker == DPT.WEEK) ? "src-dp-fade-out" : "src-dp-fade-in"}`} duration={150}>
                        <DateYear date={date} setDate={setDate} setSelectedDatePicker={setSelectedDatePicker} lang={lang} />              
                    </CSSTransition>                
                    <CSSTransition value={DPT.DECADE} className="src-dp-fade-out" duration={150}>
                        <DateDecade date={date} setDate={setDate} setSelectedDatePicker={setSelectedDatePicker} />                
                    </CSSTransition>
                </TransitionSwitch>
            </div>
        </PickerElement>
    )
}

export default DatePickerItem