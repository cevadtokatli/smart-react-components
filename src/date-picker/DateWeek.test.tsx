import React from "react"
import renderTheme from "../test/renderTheme"
import DateWeek from "./DateWeek"
import * as WSD from "../constants/date-picker/week-start-day"
import DEFAULT_LANGUAGE from "../constants/date-picker/default-language"

describe("<DateWeek />", () => {
    let props

    beforeEach(() => {
        props = {
            valueObject: {
                date:  {
                    month: 6,
                    selectedMonth: {
                        month: 6,
                        year: 2020
                    },
                    year: 2020,
                    decade: [2020, 2030]
                },
                hour: {
                    hour: 6,
                    minute: 30,
                    am: false
                }
            },
            date: {
                month: 6,
                selectedMonth: {
                    month: 6,
                    year: 2020
                },
                year: 2020,
                decade: [2020, 2030]
            },
            setValue: jest.fn(),
            weekStartDay: WSD.MONDAY,
            lang: DEFAULT_LANGUAGE
        }
    })

    it("should match snapshot", () => {
        const wrapper = renderTheme(<DateWeek {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})