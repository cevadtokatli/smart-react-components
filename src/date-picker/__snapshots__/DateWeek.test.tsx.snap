// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateWeek /> should match snapshot 1`] = `
<div>
  <ul
    class="sc-AxjAm bzDpNA"
  >
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      Mo
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      Tu
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      We
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      Th
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      Fr
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      Sa
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      Su
    </li>
  </ul>
  <ul
    class="sc-AxjAm bzDpNA"
  >
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      29
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      30
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      1
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      2
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      3
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      4
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      5
    </li>
  </ul>
  <ul
    class="sc-AxjAm bzDpNA"
  >
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      6
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      7
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      8
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      9
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      10
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      11
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      12
    </li>
  </ul>
  <ul
    class="sc-AxjAm bzDpNA"
  >
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      13
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      14
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      15
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      16
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      17
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      18
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      19
    </li>
  </ul>
  <ul
    class="sc-AxjAm bzDpNA"
  >
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      20
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      21
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      22
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      23
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      24
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      25
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      26
    </li>
  </ul>
  <ul
    class="sc-AxjAm bzDpNA"
  >
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      27
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      28
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      29
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      30
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  "
    >
      31
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      1
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      2
    </li>
  </ul>
  <ul
    class="sc-AxjAm bzDpNA"
  >
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      3
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      4
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      5
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      6
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      7
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      8
    </li>
    <li
      class="sc-AxirZ bpurbs src-date-picker-week-item src-date-picker-week-item src-date-picker-item  src-date-picker-item-inactive"
    >
      9
    </li>
  </ul>
</div>
`;
